File: gm-utils/js/src/commom/utils/integer/IntegerBox.js
export default class IntegerBox {

	constructor(i) {
		this.set(i);
	}
	set(i) {
		this.value = i;
		return this.value;
	}
	get() {
		return this.value;
	}
	inc1() {
		return this.inc(1);
	}
	inc(i) {
		return this.set(this.value+i);
	}
	dec1() {
		return this.dec(1);
	}
	dec(i) {
		return this.set(this.value-i);
	}
	toString() {
		return ""+this.value;
	}

}

File: gm-utils/src/main/java/src/commom/utils/integer/IntegerBox.java
package src.commom.utils.integer;

public class IntegerBox {
	private Integer value;

	public IntegerBox(Integer i) {
		set(i);
	}
	public Integer set(Integer i) {
		value = i;
		return value;
	}
	public Integer get() {
		return value;
	}
	public Integer inc1() {
		return inc(1);
	}
	public Integer inc(int i) {
		return set(value+i);
	}
	public Integer dec1() {
		return dec(1);
	}
	public Integer dec(int i) {
		return set(value-i);
	}
	@Override
	public String toString() {
		return ""+value;
	}

}
